### PR TITLE
persist onboarding tips visibility

### LINKED ISSUE #38: The onboarding tips should stay closed if you refresh the page
To implement this, add a new state variable `should_onboarding_tips_be_closed: boolean` to the `useGlobalStore` hook, then set that

### PR DESCRIPTION
**Pull request by mrudulpatil18**
fixes #38 

- Added `should_onboarding_tips_be_closed` & `setOnboardingTipsClosed` to global store 
- Updated `LandingHero` to use global store for onboarding tips visibility

### CODE REVIEW

File: src/components/LandingHero.tsx [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-import { CreateNewSnippetHero } from "./CreateNewSnippetHero"
 import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card"
+import { Link, useLocation } from "wouter"
+import { useGlobalStore } from "../hooks/use-global-store"
+import { CreateNewSnippetHero } from "./CreateNewSnippetHero"
+import { TypeBadge } from "./TypeBadge"
 
 export const LandingHero = () => {
-  const [visible, setVisible] = useState(true)
+  const { should_onboarding_tips_be_closed, setOnboardingTipsClosed } = useGlobalStore()
```
Comment (seveibar) [inline src/components/LandingHero.tsx:9]: use a selector with the global store- it will cause less react-rerenders (important react optimization)

**Review by seveibar — CHANGES_REQUESTED** [review]
very close! minor change

File: src/components/LandingHero.tsx (hunk 1 above)
Comment (mrudulpatil18) [reply src/components/LandingHero.tsx]: changes made

**Comment by seveibar** [comment]
format failure, biome vs code extension w/ format on save or run `bun format`

**Comment by mrudulpatil18** [comment]
> format failure, biome vscode extension w/ format on save or run `bun format`

fixed

**Review by seveibar — APPROVED** [review]

**Pull request merged**